string.go

```diff
@@ -37,12 +38,39 @@ func RandomString(size int, charset []rune) string {
 		panic("lo.RandomString: Charset parameter must not be empty")
 	}
 
-	b := make([]rune, size)
-	possibleCharactersCount := len(charset)
-	for i := range b {
-		b[i] = charset[rand.IntN(possibleCharactersCount)]
+	sb := strings.Builder{}
+	sb.Grow(size)
+	letterIdBits := int(math.Log2(float64(nearestPowerOfTwo(len(charset)))))
```

Comment:
You will have to explain the algorithm used here. Even by reading the article you quoted. I'm unsure I get it

Reply:
My English is not very good, I hope you can understand my explanation. For example, our random character set includes lowercase letters, uppercase letters, and digits, totaling 62 characters, which require 6 bits (62 -> 64 = 2^6) to represent. An int64 has 64 bits, but the random.Int63() method generates non-negative numbers, meaning the first bit is always fixed and not random, so there are only 63 random bits. Now with 63 available bits, one random character requires 6 bits, so one random.Int63() generated int64 can be used 10 times (63 / 6).

Reply:
Your English is good, no worries.

Your explanation is clear to me.

My point was to ask to comment the code with an explanation

I think you can copy-paste the one you gave here in the code.

Also, maybe one suggestion: you could add an exported method that will be in charge of producing the random number, to leave the RandomString code simpler.

Reply:
thank you